examples/internlm/README.md

```diff
@@ -0,0 +1,428 @@
+# For 20B models
+python hf_internlm_convert.py -i ./internlm-chat-20b -o ./internlm-chat-20b/smooth_internlm/sq0.5/ -sq 0.5 --tensor-parallelism 1 --storage-type fp16
+```
+
+[`build.py`](./build.py) add new options for the support of INT8 inference of SmoothQuant models.
+
+`--use_smooth_quant` is the starting point of INT8 inference. By default, it
+will run the model in the _per-tensor_ mode.
+
+Then, you can add any combination of `--per-token` and `--per-channel` to get the corresponding behaviors.
+
+Examples of build invocations:
+
+```bash
+# Build model for SmoothQuant in the _per_tensor_ mode.
+# 7B model
+python build.py --ft_model_dir=./internlm-chat-7b/smooth_internlm/sq0.5/1-gpu/ \
+                --use_smooth_quant \
+                --output_dir ./internlm-chat-7b/trt_engines/smoothquant/1-gpu
+
+# 20B model
+python build.py --ft_model_dir=./internlm-chat-20b/smooth_internlm/sq0.5/1-gpu/ \
+                --use_smooth_quant \
+                --output_dir ./internlm-chat-20b/trt_engines/smoothquant/1-gpu
+
+# OR build model for SmoothQuant in the _per_token_ + _per_channel_ mode
+# 7B model
+python build.py --ft_model_dir=./internlm-chat-7b/smooth_internlm/sq0.5/1-gpu/ \
+                --use_smooth_quant \
+                --per_token \
+                --per_channel \
+                --output_dir ./internlm-chat-7b/trt_engines/smoothquant/1-gpu
+
+# 20B model
+python build.py --ft_model_dir=./internlm-chat-20b/smooth_internlm/sq0.5/1-gpu/ \
+                --use_smooth_quant \
+                --per_token \
+                --per_channel \
+                --output_dir ./internlm-chat-20b/trt_engines/smoothquant/1-gpu
+```
+
+Note we use `--ft_model_dir` instead of `--model_dir` and `--meta_ckpt_dir` since SmoothQuant model needs INT8 weights and various scales from the binary files.
+
+Test with `run.py` or `summarize.py`:
+
+```bash
+python run.py --max_output_len=120 \
+              --input_text 'Tell me about yourself.' \
+              --tokenizer_dir ./internlm-chat-7b/ \
+              --engine_dir ./internlm-chat-7b/trt_engines/smoothquant/1-gpu
+
+python run.py --max_output_len=120 \
+              --input_text 'Tell me about yourself.' \
+              --tokenizer_dir ./internlm-chat-20b/ \
+              --engine_dir ./internlm-chat-20b/trt_engines/smoothquant/1-gpu
+
+python summarize.py --test_trt_llm --test_hf \
+                    --hf_model_location ./internlm-chat-7b \
+                    --data_type fp16 \
+                    --engine_dir ./internlm-chat-7b/trt_engines/smoothquant/1-gpu
+
+python summarize.py --test_trt_llm --test_hf \
+                    --hf_model_location ./internlm-chat-20b \
+                    --data_type fp16 \
+                    --engine_dir ./internlm-chat-20b/trt_engines/smoothquant/1-gpu
+```
+
+#### FP8 Post-Training Quantization
```

Comment:
No need to mention FP8 at current stage.

Reply:
Removed.